ooniprobe/View/TestResults/Footer/UploadFooterViewController.m

```diff
@@ -87,14 +87,17 @@ -(void)uploadMeasurements:(NSArray *)notUploaded startAt:(NSUInteger)idx{
         float progress = 0.0f;
         //The progress should consider the array size - the idx of first element to actually be uploaded
         float measurementValue = 1.0/([notUploaded count] - i);
+        MKReporterTask *task = [[MKReporterTask alloc]
```

Comment:
I guess it makes more sense to create `MKReporterTask` inside  `uploadMeasurement` rather than passing it as parameter. Or explain me this implementation.

Reply:
You want to reuse the same `task` such that you continue using the existing HTTP client, so you reuse already existing connections. Also, the same task keeps the report open as long as needed and then closes the report when you are submitting something completely different. For these two reasons, the `task` needs to have a lifecycle larger than the one of the loop. If we do that inside of `uploadMeasurement`, we lose all the improvements to performance that I mentioned above and we are as small as we used to be.

Reply:
Got it thanks!